tlatools/org.lamport.tlatools/test/tlc2/debug/ExpressionBreakpointTest.java

```diff
@@ -0,0 +1,66 @@
+package tlc2.debug;
+
+import org.eclipse.lsp4j.debug.SetBreakpointsArguments;
+import org.eclipse.lsp4j.debug.SourceBreakpoint;
+import org.junit.Test;
+import org.junit.Assert;
+
+import tlc2.output.EC;
+import tlc2.value.impl.IntValue;
+import tlc2.value.impl.LazyValue;
+import util.UniqueString;
+
+public class ExpressionBreakpointTest extends TLCDebuggerTestCase {
+
+	private static final String FOLDER = "debug";
+	private static final String RM = "ExpressionBreakpointTest";
+
+	public ExpressionBreakpointTest() {
+		super(RM, FOLDER, new String[] { "-config", RM + ".tla" }, EC.ExitStatus.SUCCESS);
+	}
+	
+	private static int getContextVal(TLCStackFrame frame, String var) {
+		Object result = frame.getContext().lookup(s -> s.getName().equals(var));
+		if (result instanceof IntValue) {
+			return ((IntValue)result).val;
+		} else if (result instanceof LazyValue) {
+			// Evade cache guard by guessing value from small range.
```

Comment:
Why is this necessary?

Reply:
I don't know how to access the value of lazy things. It seems like it is guarded fairly well.

Reply:
```java
return ((IntValue) lazy.getValue(frame.tool, frame.getS(), frame.getT(), EvalControl.Clear)).val;
```

Reply:
Even shorter:

```java
return ((IntValue) frame.unlazy(lazy)).val;
```